ONSdigital/eq-author-app
# EAR-1612-create-validation-for-count-less-than-options-available

https://collaborate2.ons.gov.uk/jira/browse/EAR-1612

### What is the context of this PR?

Prevent authors from being able to input a checkbox "count of" value greater than the number of checkboxes in the answer when creating routing or skip logic.

Example: If a checkbox answer has 3 checkboxes, and "count of" is entered as 4 by the user in the routing logic, a validation message should be shown stating that the count is greater than the number of checkboxes in the answer.

This should apply to Skip conditions too.

### How to review

- create a Checkbox Answer with any number of checkboxes (Can include "Other" and "OR" checkboxes)
- in this answer create a routing Rule
- Select this answer as the "IF" - Left hand side of the rule
- Select "COUNTOF" for the Match drop down
- Select any option except "MORE THAN" for the secondary condition
- Add a greater number (than the number of checkboxes you created) - for the input
- This should Generate the following error message - "Count is greater than the number of checkboxes in the answer"

For the "MORE THAN" secondary condition:
Add the SAME OR greater number (than the number of checkboxes you created) - for the input
To generate the same error message

Do the same for Skip condition Logic
